pyteal/ast/subroutine.py

```diff
@@ -610,32 +611,73 @@ def method_signature(self, overriding_name: str = None) -> str:
         return f"{overriding_name}({','.join(args)}){self.type_of()}"
 
     def method_spec(self) -> sdk_abi.Method:
-        skip_names = ["return", "output"]
+        desc: str = ""
+        arg_descs: dict[str, str] = {}
+        return_desc: str = ""
+        args: list[dict[str, str]] = []
 
-        args = [
-            {
-                "type": str(abi.type_spec_from_annotation(val)),
-                "name": name,
-            }
-            for name, val in self.subroutine.annotations.items()
-            if name not in skip_names
-        ]
+        if self.subroutine.implementation.__doc__:
+            docstring = parse_docstring(self.subroutine.implementation.__doc__)
 
-        spec = {
-            "name": self.name(),
-            "args": args,
-            "returns": {"type": str(self.type_of())},
-        }
+            # Combine short and long descriptions with newline
+            method_descriptions: list[str] = []
+            if docstring.short_description:
+                method_descriptions.append(docstring.short_description)
+            if docstring.long_description:
+                method_descriptions.append(docstring.long_description)
 
-        if self.subroutine.implementation.__doc__ is not None:
-            spec["desc"] = " ".join(
+            method_desc = "\n\n".join(method_descriptions)
+
+            # Turn double new line into single, replacing single newline with space
+            desc = "\n".join(
                 [
-                    i.strip()
-                    for i in self.subroutine.implementation.__doc__.split("\n")
+                    i.replace("\n", " ").strip()
+                    for i in method_desc.split("\n\n")
                     if not (i.isspace() or len(i) == 0)
```

Comment:
minor trick, I think you might like 
```suggestion
                    if i.strip()
```
for it covers both check non space and len > 0